File: gohan-main/src/main/java/com/dahan/gohan/repository/GohanDependency.java
package com.dahan.gohan.repository;
/* ************************************************************************
 *
 * Copyright (C) 2020 dahan All rights reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * ************************************************************************/

/*
 * Creates on 2020/12/9.
 */

import com.dahan.gohan.StringUtils;
import org.eclipse.aether.artifact.DefaultArtifact;
import org.eclipse.aether.graph.Dependency;
import org.eclipse.aether.util.graph.transformer.ConflictResolver;

/**
 * 依赖基础信息类
 *
 * @author kevin
 */
public final class GohanDependency
{

    private String groupId;

    private String artifactId;

    private String version;

    private String scope = "compile";

    private String classifier = "classifier";

    private String extension = "jar";

    private Dependency dependency;

    public GohanDependency(String groupId, String artifactId, String version)
    {
        this(groupId, artifactId, version, null, null);
    }

    public GohanDependency(String groupId, String artifactId, String version, String classifier)
    {
        this(groupId, artifactId, version, classifier, null);
    }

    public GohanDependency(String groupId, String artifactId, String version, String classifier, String scope)
    {
        this.groupId = groupId;
        this.artifactId = artifactId;
        this.version = version;

        if (StringUtils.isNotEmpty(scope))
        {
            this.scope = scope;
        }

        if (StringUtils.isNotEmpty(classifier))
        {
            this.classifier = classifier;
        }

        DefaultArtifact defaultArtifact = new DefaultArtifact(groupId, artifactId, classifier, extension, version);

        dependency = new Dependency(defaultArtifact, this.scope);

    }

    public String getGroupId()
    {
        return groupId;
    }

    public void setGroupId(String groupId)
    {
        this.groupId = groupId;
    }

    public String getArtifactId()
    {
        return artifactId;
    }

    public void setArtifactId(String artifactId)
    {
        this.artifactId = artifactId;
    }

    public String getVersion()
    {
        return version;
    }

    public void setVersion(String version)
    {
        this.version = version;
    }

    public String getScope()
    {
        return scope;
    }

    public void setScope(String scope)
    {
        this.scope = scope;
    }

    public Dependency toEclipseDependency()
    {
        return dependency;
    }

}

File: gohan-repository/src/main/java/com/dahan/gohan/repository/GohanDependency.scala
package com.dahan.gohan.repository

/* ************************************************************************
 *
 * Copyright (C) 2020 dahan All rights reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License")
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * ************************************************************************/

/*
 * Creates on 2020/12/9.
 */

import org.eclipse.aether.artifact.DefaultArtifact
import org.eclipse.aether.graph.Dependency

/**
 * 依赖基础信息类
 *
 * @author tiansheng
 */
class GohanDependency {

  private var groupId: String = _

  private var artifactId: String = _

  private var version: String = _

  private var scope: String = "compile"

  private var classifier: String = _

  /**
   * eclipse dependency object.
   */
  private var dependency: Dependency = _

  def this(coords: String, classifier: String = null) {
    this()
    val coordsArray: Array[String] = coords.split(":")
    pseudo(coordsArray(0), coordsArray(1), coordsArray(2), null)
  }

  def this(groupId: String, artifactId: String, version: String) {
    this()
    pseudo(groupId, artifactId, version)
  }

  def this(groupId: String, artifactId: String, version: String, classifier: String) {
    this()
    pseudo(groupId, artifactId, version, classifier)
  }

  def this(groupId: String, artifactId: String, version: String, classifier: String, scope: String) {
    this()
    pseudo(groupId, artifactId, version, classifier, scope)
  }

  /**
   * 伪构造器
   *
   * @param version    版本号
   * @param classifier 分类
   * @param scope      范围
   */
  private def pseudo(groupId: String, artifactId: String, version: String, classifier: String = null, scope: String = null): Unit = {
    this.groupId = groupId
    this.artifactId = artifactId
    this.version = version

    if (scope != null) {
      this.scope = scope
    }

    if (this.classifier == null) {
      this.dependency = new Dependency(new DefaultArtifact(s"$groupId:$artifactId:$version"), this.scope)
    } else {
      this.classifier = classifier
      this.dependency = new Dependency(new DefaultArtifact(groupId, artifactId, classifier, "jar", version), this.scope)
    }
  }

  def getGroupId: String = groupId

  def getArtifactId: String = artifactId

  def getVersion: String = version

  def getDependency: Dependency = dependency

}